worldedit-core/src/main/java/com/sk89q/worldedit/session/SessionManager.java

```diff
@@ -192,6 +183,22 @@ public synchronized LocalSession get(SessionOwner owner) {
         return session;
     }
 
+    private boolean shouldBoundLimit(int defaultLimit, int maxLimit, int currentLimit) {
+        // If the maximum isn't infinite, ensure the default isn't larger than it.
+        if (maxLimit > -1) {
+            if (defaultLimit < 0) {
+                if (currentLimit < 0 || currentLimit > maxLimit) {
+                    return true;
+                }
+            } else {
+                if (currentLimit == -1 || currentLimit > maxLimit) {
```

Comment:
Something is messed up here... one limit is tested == -1, the other limit is tested < 0, yet another is tested > -1.
Also, defaultLimit is never checked for being larger than the maximum, so the comment is wrong.

Reply:
config loading checks `Math.max(-1, config.getInt(...` so if it's less than zero, it must be -1. but yea that relies on the config's check. we can make it explicit here. default CAN be larger than max, IF the player has override perms. we actually are bounding current to max here. this method is only called if the player doesn't have override perms.